src/handler/job/monitorRanks.ts

```diff
@@ -20,4 +39,52 @@ export class MonitorRanks {
             processUser(u, guilds);
         });
     }
+
+    @DiscordEvent("guildMemberAdd")
+    async initUserRank(member: GuildMember) {
+        logger.info(`${member.displayName} joined ${member.guild.name}`);
+        try {
+            const userData = await getUser(config.apiKey, member.id);
+            const guild = await getGuild(member.guild.id);
+            updateRoleAndNickname(userData, guild, member);
+            sendMessage(
+                member,
+                `Successfully verified user ${member.user.tag}!`,
+                Color.GREEN
+            );
+        } catch (e) {
+            if (e instanceof ApiError && e.code === 2) {
+                sendMessage(
+                    member,
+                    `Your discord account is not verified with Final Earth.
+                Please visit [here](https://www.finalearth.com/discord) and follow the instructions.`,
+                    Color.RED
+                );
+            } else {
+                logger.error(e);
+            }
+        }
+    }
+
+    @DiscordEvent("guildMemberRemove")
+    async memberLeft(member: GuildMember | PartialGuildMember) {
```

Comment:
will this cause issues if this doesn't complete successfully for some reason?

Reply:
it should be fine, it wont remove the guild id for the user, but that doesn't really matter.

it'll just mean that the application will waste time trying to fetch guilds the user isnt in.

Reply:
ok, we can just add a cleanup job later that removes dead users then